### DISCUSSION TITLE
Configuration of different usage scenarios of dev mode

### DISCUSSION THREAD

**Discussion by misitebao** [wailsapp/wails#1384] [General]
I think the configuration in `wails dev` mode is mainly divided into four cases.

| frontend:dev | frontend:dev:watcher | frontend:dev:serverUrl |                                          e.g.                                           |
| :----------: | :------------------: | :--------------------: | :-------------------------------------------------------------------------------------: |
|      ✔       |                      |                        |                    "frontend:dev": "npm run build -mode development"                    |
|              |          ✔           |                        |            "frontend:dev:watcher": "npm run build -mode development --watch"            |
|              |          ✔           |           ✔            | "frontend:dev:watcher": "npm run dev","frontend:dev:serverUrl": "http://localhost:3000" |
|              |                      |                        |    1.prompt for configuration errors, 2. use the value specified by `frontend:build`.     |

The first three scenarios are bound to exist. For the fourth scenario, I prefer the first solution, which can separate `dev` and `build`.

**Comment by leaanthony** [comment]
For development, vite has always been a preferred solution as it does everything 99% of people need. Our initial reload trigger for frontend changes were a stopgap. I'm in favour of dropping it altogether and providing docs on how to do this out-of-band. Trying to work out when an external process has finished so you can safely reload is absolutely non trivial, if not impossible. Hitting the `/wails/reload` endpoint triggers all the logic in the system: reload app, reload connected browsers. As we have no telemetry, we have no idea if the majority of the frontend dev is in a browser anyway so maybe just manually pressing reload is fine? I'm not even sure we should be supporting the watcher command at this point: it's trivial to do that yourself and not have the headache of something in Wails going wrong whilst performing an `exec`.

TL;DR;

Let's remove non-vite Dev tooling, but make that information available to Devs who need it

@stffabi  thoughts?

**Reply by stffabi** [reply]
I would support removing non-vite dev tooling 👍, making the integration in Wails cleaner and allows us removing some special handlings.

I would vote for having just a simple reload mechanism in wails dev for those of us that are not using vite and are not developing in the browser (like me 😄). 
Something like flutter would be enough: Pressing r/R when running Wails Dev to (r) reload frontend or (R) to reload the whole app.

**Reply by leaanthony** [reply]
I agree, but do wonder how that might affect current remote templates.
